packages/classes/src/Payload.js

```diff
@@ -26,6 +26,24 @@ class Payload {
 		})
 	}
 
+	/**
+	 * Checks whether an object is an instance of a Payload.
+	 *
+	 * This uses duck typing, rather than instanceof, since we can't rely on the prototype chain
+	 * matching across packages and versions.
+	 *
+	 * @param  {Object} obj The object being tested.
+	 * @return {Boolean}     The result of the duck test.
+	 */
+	static isPayload = (obj) => (
+		Object.prototype.hasOwnProperty.call(obj, 'data')
```

Comment:
Seems fine! Could it be written as `obj.prototype.hasOwnProperty('data')`? Honestly not sure, assume there's a reason it's written this way. Might be more succinctly written as an `every` call on an array. Could maybe get the members from something like `Object.getOwnPropertyNames(Object.getPrototypeOf(Payload))`.

Reply:
oh interesting -- eslint [yelled at me](https://eslint.org/docs/rules/no-prototype-builtins) when I had done `obj.hasOwnProperty()` but I didn't try `obj.prototype.*`.

I also like the idea of having this automatically update when we add new properties using something like getOwnPropertyNames and iterating.  I'm going to give that a shot!  Thanks!

Reply:
`Object.getOwnPropertyNames(Object.getPrototypeOf(Payload))` had a lot of cruft that went away when I replaced it with `Object.getOwnPropertyNames(new Payload())`

I preferred the idea of not instantiating for this (since that's more re-usable especially for interface duck typing) but I still think this path is better than what I had originally.

Reply:
Makes sense. Agree on not liking having to new an object. If we have to do this for every packet, may get expensive. Maybe use a static list instead.

Otherwise, if you want to save a few characters, may be able rewrite what you have as:
```js
Object.getOwnPropertyNames(new Payload()).every(Object.prototype.hasOwnProperty.bind(obj))
```

Reply:
Oh how quickly I forget about efficiency.

Possible paths:

1. Leave as is and optimize later
2. Look into caching the static "archetype object" and only creating it the first time `isPayload` is called
3. Static list of properties (similar to what was before, but building off of this with a smarter syntax)

1 feels like the wrong choice
2 feels like it might be a bit dirty
3 would require the developer to remember to add properties to the list to avoid opening some kind of obscure edge case, but these objects ought not to often change and frankly does not represent much of a threat since these checks are generally defensive to begin with.

Reply:
3 could also be made to break a test.

Reply:
Great point -- SOLD!

I will spend like 5 minutes checking if maybe there is some secret approach that uses prototypes to get the list of items, but then I'll just implement 3 (also, the new IAppliance PR will be updated to whatever we do here)

Reply:
This whole thing reminded me of this [super long](https://overreacted.io/how-does-react-tell-a-class-from-a-function/) post about React, which you don't need to read, and how I solved comparing this in one case elsewhere in this project, which was to check `obj.constructor.name`, which does return `'Payload'`.

Reply:
Specifically [here](https://github.com/tvkitchen/tv-kitchen/blob/master/lib/utils/jest.setup.js#L9).

Reply:
Good call -- and that would work for this PR for sure (though it wouldn't help for the IAppliance case since appliances will generally be extended).  For IAppliance the optimization is way less important though, since those get checked / constructed during project setup (vs payloads which might be checked thousands of times per second).

I don't think there's any need for the approach to be the same for IAppliance as it is here, just flagging it for kicks.

Reply:
Quick note: love the code golf bind example; sticking with the explicit call since I think that's a bit more legible for folks unfamiliar with bind (aka me before you pointed out that possibility lol)

Reply:
Just to note where I landed on this: I followed path 3, of having an explicit list of items.  In order to test the completeness of that list I opted to store it as a static "duckTypeProperties" attribute of the class.

Just for kicks I decided to try having that static attribute populated with the `getOwnPropertyNames` approach, expecting an error.

It did not error!

That property is NOT an official part of the class API so we can remove it later.  Chris brought up the idea of maybe using a marker (e.g. an `isPayload` flag set via constructor) to short cut, but I'm going to punt on that for now since this is more robust and verifies the actual functionality of the payload.

Moving forward, we can circle back and adjust implementation to the other options in future; the important thing is that the API now has an isPayload / isPayloadArray method.